syntax = "proto3";

option go_package = "internal/generated/dbus/consume/uziprocessed";

message UziSplitted {
    string uzi_id = 100;
    repeated string pages_id = 200;
}

message UziProcessed {
    // из-за коллизии имен с grpc, пришлось использовать вложенные сообщения
    message Node {
        bool ai = 100;
        double tirads_23 = 400;
        double tirads_4 = 500;
        double tirads_5 = 600;
    }
    
    message Segment {
        string image_id = 100;
        bytes contor = 200;
        double tirads_23 = 300;
        double tirads_4 = 400;
        double tirads_5 = 500;
    }
    
    message NodeWithSegments {
        Node node = 100;
        repeated Segment segments = 200;
    }

    string uzi_id = 100;
    repeated NodeWithSegments nodes_with_segments = 200;
}